dace/sdfg/analysis/cutout.py

```diff
@@ -213,11 +227,24 @@ def singlestate_cutout(cls,
             memlet = edge.data
             if memlet.data in cutout.arrays:
                 continue
-            new_desc = sdfg.arrays[memlet.data].clone()
-            cutout.add_datadesc(memlet.data, new_desc)
+            dataname = memlet.data
+            if '.' in dataname:
+                # This is an access to a struct memeber, which typically happens for the memlets between an access node
+                # pointing to a struct (or view thereof), and a view pointing to the member. Assert that this is indeed
+                # the case (i.e., only one '.' is found in the name of the data being accessed), and if so, clone the
+                # struct (or struct view) data descriptor instad.
+                parts = dataname.split('.')
+                if len(parts) == 2:
+                    dataname = parts[0]
+                else:
+                    raise RuntimeError('Attempting to add invalid multi-nested data ' + memlet.data + ' to a cutout')
```

Comment:
There is a get root data descriptor function as far as I'm aware. Please use it.

Reply:
I can only find one defined as a property on `AccessNode`s (`AccessNode::root_data`). However, that appears to be doing something different than what this portion here attempts to do. `AccessNode::root_data` always grabs `dataname.split('.')[0]`, no matter how long the sub-member access chain is, whereas here we specifically want to make sure that there is only one nesting level.